consensus/state_processing/src/genesis.rs

```diff
@@ -123,6 +123,19 @@ pub fn initialize_beacon_state_from_eth1<E: EthSpec>(
         // Remove intermediate Deneb fork from `state.fork`.
         state.fork_mut().previous_version = spec.electra_fork_version;
 
+        // iterate over an empty pending_deposits list here and increase balance
+        let pending_deposits = state.pending_deposits()?.clone();
+        for pending_deposit in pending_deposits.into_iter() {
+            if let Some(validator_index) = state.pubkey_cache().get(&pending_deposit.pubkey) {
+                state
+                    .balances_mut()
+                    .get_mut(validator_index)
+                    .ok_or(Error::UnknownValidator(validator_index))?
+                    .safe_add(pending_deposit.amount)?;
+            }
+        }
+        *state.pending_deposits_mut()? = List::empty();
```

Comment:
My understanding was this applies pending_deposits to the state during initialization if the state contains pending_deposits. Here's the part of the [spec](https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.8/specs/electra/beacon-chain.md#testing)
```python
for deposit in state.pending_deposits:
        validator_index = ValidatorIndex(validator_pubkeys.index(deposit.pubkey))
        increase_balance(state, validator_index, deposit.amount)
    state.pending_deposits = []
```

Reply:
Ohh right, I was confused because we are calling `upgrade_to_electra` immediately prior to this, which sets the `pending_deposits` to the empty list:

https://github.com/sigp/lighthouse/blob/6d10456912b3c39b8a8c9089db76e8ead20608a0/consensus/state_processing/src/genesis.rs#L120

https://github.com/sigp/lighthouse/blob/c042dc14d74352512b7632e0ee6ec07f1aa26b3a/consensus/state_processing/src/upgrade/electra.rs#L140

I think we might need to move the `apply_deposit` logic _after_ upgrading to Electra, so that the `pending_deposits` list gets populated. Currently it happens here:

https://github.com/sigp/lighthouse/blob/6d10456912b3c39b8a8c9089db76e8ead20608a0/consensus/state_processing/src/genesis.rs#L36-L43

Or maybe we just delete the whole `initialize_beacon_state_from_eth1` function and say we don't support starting from eth1 genesis any more :sweat_smile:

It's interesting that the EF tests for `initialize_beacon_state_from_eth1` already pass. Maybe processing the deposits with the pre-Electra logic is actually equivalent, and we could just delete the code here that loops over `ppending_deposits`?

Reply:
Yeah i'm okay with that. It seemed redundant to me initially but i just implemented it for completeness.
I'm fine with removing it

Reply:
Removed in d5ec058